File: Python/program7.py
ERR_INVALID_AGE = -1

def CalculateTicket(age):
    if age < 0:
        return ERR_INVALID_AGE

    if 0 <= age <= 5:
        return 0
    elif 5 < age <= 18:
        return 700
    elif 18 < age <= 50:
        return 999
    elif age > 50:
        return 500


def main():
    try:
        age = int(input("Please enter your age : "))
        ticket_price = CalculateTicket(age)

        if ticket_price == ERR_INVALID_AGE:
            print("Unable to calculate the ticket price as age is invalid")
        else:
            print(f"Your ticket price is : {ticket_price}")
    except ValueError:
        print("Invalid input! Please enter a valid number for age.")


if __name__ == "__main__":
    main()

File: JAVA/program7.java
import java.util.*;
class CalculateTicket
{
    public static final int ERR_INVALID_AGE = -1;

    public int TicketPrice( int iAge)
    {
        if(iAge < 0)
        {
            return ERR_INVALID_AGE;
        }

        else if((iAge >= 0) && (iAge <= 5))
        {
            return 0;
        }

        else if((iAge > 5) &&(iAge <= 18))
        {
            return 700;
        }

        else if((iAge > 18) && (iAge <= 50))
        {
            return 999;
        }

        else if(iAge > 50)
        {
            return 500;
        }

        return ERR_INVALID_AGE;

    }
}

class program7
{
    public static void main(String[] A)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0, iRet = 0;

        System.out.println("Enter your age :");
        iValue = sobj.nextInt();

        CalculateTicket cobj = new CalculateTicket();
        iRet = cobj.TicketPrice(iValue);

        if(iRet == cobj.ERR_INVALID_AGE)
        {
            System.out.println("Unable to count ticket price because your age is invalid:");
        }
        else
        {
            System.out.println("Your ticket price is : "+ iRet);
        }

    }
}